src/cmds/scripts/win_postinstall.py

```diff
@@ -0,0 +1,632 @@
+            "#log_directory = 'pg_log'",
+            "log_directory = 'pg_log'")
+        # change logging filename
+        replace(
+            postgresql_file_path,
+            "#log_filename = 'postgresql-%Y-%m-%d_%H%M%S.log'",
+            "log_filename = 'pbs_dataservice_log.%a'")
+        # change logging truncate
+        replace(
+            postgresql_file_path,
+            "#log_truncate_on_rotation = off",
+            "log_truncate_on_rotation = on")
+        # change logging rotation age
+        replace(
+            postgresql_file_path,
+            "#log_rotation_age = 1d",
+            "log_rotation_age = 1440")
+        # change logging format to add timestamp
+        replace(
+            postgresql_file_path,
+            "#log_line_prefix = ''",
+            "log_line_prefix = '%t'")
+        print "postgresql file Successfully configured"
+    else:
+        print "postgresqlfile is not available to configure"
+        exit(7)
+    # Configuring pg_hba.conf file
+    pg_hba_file_path = os.path.join(
+        pbs_path, "home", "datastore", "pg_hba.conf")
+    if (os.path.exists(pg_hba_file_path)):
+        replace(
+            pg_hba_file_path,
+            "# IPv4 local connections:",
+            "# IPv4 connections:")
+        # Change localaddress values for all users
+        original_str = "host    all             all             " + \
+            "127.0.0.1/32            trust"
+        replacement_str = "host    all         all         " + \
+            "127.0.0.1/32          md5\nhost       all         " + \
+            "all        0.0.0.0/0      md5"
+        replace(pg_hba_file_path, original_str, replacement_str)
+        # Change user specific config
+        original_str = "# Allow replication connections from localhost,\
+        by a user with the"
+        replacement_str = "host     all         all         localhost\
+        md5"
+        replace(pg_hba_file_path, original_str, replacement_str)
+        print "pg_hba file Successfully configured"
+    else:
+        print "pg_hba file is not available to configure"
+        exit(8)
+
+
+def InstallDataService(username, password):
+    pbs_pathquote = "\"" + pbs_path
+    # Create server_priv directory
+    server_priv_path = os.path.join(pbs_path, 'home', 'server_priv')
+    if not os.path.exists(server_priv_path):
+        os.makedirs(server_priv_path)
+    # Creation of db_user file at PBS_HOME\server_priv directory
+    dbuserfile = os.path.join(pbs_path, 'home', 'server_priv', 'db_user')
+    with open(dbuserfile, 'w') as db:
+        db.write(username)
+        db.close()
+    # Creation of createdb.bat script at PBS_EXEC\etc directory
+    DBpath = "\"" + os.path.join(pbs_path, "home", "datastore") + "\""
+    createdbfile = os.path.join(pbs_path, "exec", "etc", "createdb.bat")
+    ErrChk = "IF %ERRORLEVEL% NEQ 0 GOTO ERR\r\n"
+    with open(createdbfile, 'w') as crtdb:
+        buffer = "@echo off \r\n"
+        crtdb.write(buffer)
```

Comment:
Rather than writing each line in to file can we use some kind a template then change required value in template, it will help user better readability on what is going in this dynamic script.